src/web/ng/SubscriptionContext.cpp

```diff
@@ -50,24 +51,31 @@ SubscriptionContext::SubscriptionContext(
 {
 }
 
+SubscriptionContext::~SubscriptionContext()
+{
+    ASSERT(disconnected_, "SubscriptionContext must be disconnected before destroying");
```

Comment:
That's UB if the exception is thrown, but I think we should be fine.

Reply:
I added `WithMockAssertNoThrow` fixture for this case in tests.